idb/postgres/postgres.go

```diff
@@ -658,6 +658,9 @@ func buildTransactionQuery(tf idb.TransactionFilter) (query string, whereArgs []
 	if tf.RekeyTo != nil && (*tf.RekeyTo) {
 		whereParts = append(whereParts, "(t.txn -> 'txn' -> 'rekey') IS NOT NULL")
 	}
+	if tf.ReturnRootTxnsOnly {
+		whereParts = append(whereParts, "t.txid IS NOT NULL")
+	}
 
 	// If returnInnerTxnOnly flag is false, then return the root transaction
 	if !tf.ReturnInnerTxnOnly {
```

Comment:
This is also needed to make sure we don't try to reference `root`:
```suggestion
	if !tf.ReturnInnerTxnOnly && !tf.ReturnRootTxnsOnly {
```